src/App.js

```diff
@@ -5,13 +5,35 @@ import Messages from './Components/Messages';
 import OnlineUsers from './Components/OnlineUsers';
 import MessageForm from './Components/MessageForm';
 import './css/main.css';
+import io from 'socket.io-client'
+let socket = io()
 
 class App extends Component {
     constructor(props) {
         super(props)
         this.state = ({
             username: "user01",
-            online: ["user01", "user02"]
+            online: ["user01", "user02"],
+            message: null,
+            messages: []
+        });
+    }
+
+    componentDidMount(){
+        this.handleMessageUpdate()
+    }
+
+    handleSubmit(event) {
+        event.preventDefault()
+        socket.emit('chat message', this.state.message)
+            this.setState({ message: '' })
+        }
+
+    handleMessageUpdate() {
+        socket.on('chat message', (msg) => {
+        var currentMessages = this.state.messages
+        currentMessages.push(msg);
```

Comment:
Yeah I agree with your Slack comment about this. We should think of a better way of doing it. But for now this works.

Reply:
Does the function body need to be indented here?

Reply:
Yes, good spot!